extensions/binding/org.eclipse.smarthome.binding.onewire/src/main/java/org/eclipse/smarthome/binding/onewire/internal/handler/TemperatureSensorThingHandler.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.binding.onewire.internal.handler;
+
+import static org.eclipse.smarthome.binding.onewire.internal.OwBindingConstants.*;
+
+import java.util.Collections;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.binding.onewire.internal.OwDynamicStateDescriptionProvider;
+import org.eclipse.smarthome.binding.onewire.internal.OwException;
+import org.eclipse.smarthome.binding.onewire.internal.device.DS18x20;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+
+/**
+ * The {@link TemperatureSensorThingHandler} is responsible for handling temperature sensors
+ *
+ * @author Jan N. Klug - Initial contribution
+ */
+@NonNullByDefault
+public class TemperatureSensorThingHandler extends OwBaseThingHandler {
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(THING_TYPE_TEMPERATURE);
+
+    public TemperatureSensorThingHandler(Thing thing,
+            @Nullable OwDynamicStateDescriptionProvider dynamicStateDescriptionProvider) {
+        super(thing, dynamicStateDescriptionProvider);
+    }
+
+    @Override
+    public void initialize() {
+        super.initialize();
+
+        sensors.add(new DS18x20(sensorIds.get(0), this));
```

Comment:
`sensorIds.get(0)` might fail if no sensor id was configured for this thing. See also the other Thing handlers where this can cause issues.

edit: I just saw that the framework will protect you in case a required parameter is missing. In case of a "bad" id (like `id="abcd"`)  I actually get the expected `java.lang.IndexOutOfBoundsException: Index: 0, Size: 0`.

Reply:
this will be solved by the changes i made above to the DigitalIOThingHandler intialization code (then this will never be called if the ids are invalid).